### ISSUE TITLE
[Python] Can I train each layers separately?

### ISSUE THREAD

**Issue by bowenYu** [dmlc/mxnet#344]
For example, in the /mxnet/example/minist/mlp.py, 

	data = mx.symbol.Variable('data')
	fc1 = mx.symbol.FullyConnected(data = data, name='fc1', num_hidden=128)
	act1 = mx.symbol.Activation(data = fc1, name='relu1', act_type="relu")
	fc2 = mx.symbol.FullyConnected(data = act1, name = 'fc2', num_hidden = 64)
	act2 = mx.symbol.Activation(data = fc2, name='relu2', act_type="relu")
	fc3 = mx.symbol.FullyConnected(data = act2, name='fc3', num_hidden=10)
	mlp = mx.symbol.Softmax(data = fc3, name = 'mlp')

it has three hidden layer. Is it possbile to train the hidden layers using some unlabeled data, and then using some labeled ones to fine-tune the whole net?

Maybe it is similar with this issue: Is it possible to parallize differen layers on different GPUs (https://github.com/dmlc/mxnet/issues/338)? But I'm not sure how to do it in my code. Thanks.

**Comment by pluskid** [comment]
There is also a related discussion on adding attributes to symbols (#267). One reason is to support this kind of manipulation (e.g. `Readonly` or `Frozen` attribute to some symbols). While there is no specific plan for adding that feature yet. Currently I think what you can do is to hack into `model.py` and manually skip updating the gradients corresponding to the symbols that you want to skip, around here: https://github.com/dmlc/mxnet/blob/master/python/mxnet/model.py#L312

**Comment by antinucleon** [comment]
Next PR will contain BlockGradientOp, which can be used to block gradient
backprop

**Comment by pluskid** [comment]
`BlockGrad` is already in master. Though I feel it a bit hacky way to do this as in order to block gradient, one need to change the network architecture by inserting that block-gradient operator into the layers.
